samples/trusted-ai/trusted-ai.py

```diff
@@ -35,56 +26,43 @@ def trusted_ai(
         clip_values='(0, 1)',
         nb_classes='2',
         input_shape='(1,3,64,64)'):
-    objectiveConfig = {
-      "type": "maximize",
-      "goal": goal,
-      "objectiveMetricName": "accuracy",
-      "additionalMetricNames": []
-    }
-    algorithmConfig = {"algorithmName" : "random"}
-    parameters = [
-      {"name": "--dummy", "parameterType": "int", "feasibleSpace": {"min": "1", "max": "2"}},
-    ]
-    rawTemplate = {
-      "apiVersion": "batch/v1",
-      "kind": "Job",
-      "metadata": {
-         "name": "{{.Trial}}",
-         "namespace": "{{.NameSpace}}"
-      },
-      "spec": {
-        "template": {
-          "spec": {
-            "restartPolicy": "Never",
-            "containers": [
-              {"name": "{{.Trial}}",
-               "image": "aipipeline/gender-classification:latest",
-               "command": [
-                   "python", "-u", "gender_classification_training.py", "--data_bucket", "mlpipeline", "--result_bucket", "mlpipeline"
-               ]
-              }
-            ],
-            "env": [{'name': 'S3_ENDPOINT', 'value': 'minio-service.kubeflow:9000'}]
-          }
+    job_manifest = {
+        "apiVersion": "batch/v1",
+        "kind": "Job",
+        "metadata": {
+            "name": "trusted-ai-train-job",
+            "namespace": namespace
+        },
+        "spec": {
+            "ttlSecondsAfterFinished": 100,
+            "template": {
+                "metadata": {
+                    "annotations": {
+                        "sidecar.istio.io/inject": "false"
+                    }
+                },
+                "spec": {
+                    "restartPolicy": "Never",
+                    "containers": [
+                        {"name": "classification-training",
+                         "image": "aipipeline/gender-classification:latest",
+                         "command": [
+                             "python", "-u", "gender_classification_training.py", "--data_bucket", "mlpipeline",
+                             "--result_bucket", "mlpipeline"
+                         ],
+                         "env": [{'name': 'S3_ENDPOINT', 'value': 'minio-service.kubeflow:9000'}]
+                         }
+                    ],
+                }
+            }
         }
-      }
     }
-    trialTemplate = {
-      "goTemplate": {
-        "rawTemplate": json.dumps(rawTemplate)
-      }
-    }
-    katib_run = katib_experiment_launcher_op(
-        experiment_name=name,
-        experiment_namespace=namespace,
-        parallel_trial_count=parallelTrialCount,
-        max_trial_count=maxTrialCount,
-        objective=str(objectiveConfig),
-        algorithm=str(algorithmConfig),
-        trial_template=str(trialTemplate),
-        parameters=str(parameters),
-        experiment_timeout_minutes=experimentTimeoutMinutes,
-        delete_finished_experiment=deleteAfterDone)
+    train_step = dsl.ResourceOp(
+        name="trust-ai-train-step",
```

Comment:
Name here is not a pipeline param. How to avoid name collision? It does not seem to handle itself, though the documentation says so.